catalyst-gateway/bin/src/cardano/mod.rs

```diff
@@ -524,7 +523,16 @@ impl SyncTask {
         self.dispatch_event(event::ChainIndexerEvent::SyncLiveChainStarted);
 
         self.start_immutable_followers();
-        self.dispatch_event(event::ChainIndexerEvent::SyncImmutableChainStarted);
+        // IF there is only 1 chain follower spaw, then the immutable state already indexed and
+        // filled in the db.
+        if self.sync_tasks.len() == 1 {
+            if !immutable_follower_has_first_reached_tip() {
```

Comment:
Do not do it this way.
Its not necessary.  And its also potentially racy because you are testing and then setting an atomic, non atomically. Just have a `set_follower_??_reached_tip()` function that does this:

``` rust
pub(crate) fn set_follower_immutable_reached_tip() {
    if  !INITIAL_IMMUTABLE_FOLLOWER_TIP_REACHED.swap(true, Ordering::Relaxed) {
        info!("Follower has reached IMMUTABLE TIP for the first time");
    }
}
```
It's very bad practice to do non atomic compares and sets with atomic variables.  This way dragons lie.

Reply:
updated